standard_sequence/classy_sequence.py

```diff
@@ -861,263 +318,528 @@ def _do_molasses_tof_sequence(self, t, reset_mot = False):
         t = self.do_molasses(t, dur=shot_globals.bm_time, close_all_shutters=True)
 
+        # TODO: does making this delay longer make the background better when using UV?
+        t += 7e-3
+        t = self.TweezerLaser_obj.ramp_power(
+            t, dur=TweezerLaser.CONST_TWEEZER_RAMPING_TIME, final_power=1
+        )
+        # ramp to full power and parity projection
+        if shot_globals.do_parity_projection_pulse:
+            _, t_aom_start = self.D2Lasers_obj.parity_projection_pulse(
+                t, dur=shot_globals.bm_parity_projection_pulse_dur
+            )
+            # if doing parity projection, synchronize with power ramp
+            t = t_aom_start
+            t += shot_globals.bm_parity_projection_pulse_dur
+
+        # t = self.do_molasses(t, dur=shot_globals.bm_time, close_all_shutters=True)
+        # t += shot_globals.bm_time
+
+        t = self.ramp_to_imaging_parameters(t)
+
+        if shot_globals.do_robust_loading_pulse:
+            # additional cooling, previously referred to as "robust_loading"
+            # sometimes don't use this when tweezer debugging is needed?
+            t, _ = self.D2Lasers_obj.do_pulse(
+                t,
+                shot_globals.bm_robust_loading_pulse_dur,
+                ShutterConfig.IMG_FULL,
+                shot_globals.img_ta_power,
+                shot_globals.img_repump_power,
+                close_all_shutters=True,
+            )
+
+        assert (
+            shot_globals.img_tof_imaging_delay > D2Lasers.CONST_MIN_SHUTTER_OFF_TIME
+        ), "time of flight needs to be greater than CONST_MIN_SHUTTER_OFF_TIME"
+        t += shot_globals.img_tof_imaging_delay
+
         return t
 
+    def tweezer_modulation(self, t, label="sine"):
+        pass
+
     def rearrange_to_dense(self, t):
         pass
 
     def image_tweezers(self, t, shot_number):
-        # TODO: do_kinetix_imaging needs to include pre_imaging ramps
-        t = self.do_kinetix_imaging(t, shot_number=1)
+        t = self.ramp_to_imaging_parameters(t)
+        if shot_number == 1:
+            t = self.do_kinetix_imaging(
+                t, close_all_shutters=shot_globals.do_shutter_close_after_first_shot
+            )
+        if shot_number == 2:
+            # pulse for the second shots and wait for the first shot to finish the
+            # first reading
+            kinetix_readout_time = shot_globals.kinetix_roi_row[1] * 4.7065e-6
+            # need extra 7 ms for shutter to close on the second shot
+            # TODO: is shot_globals.kinetix_extra_readout_time always zero? Delete if so.
+            t += kinetix_readout_time + shot_globals.kinetix_extra_readout_time
+            t = self.do_kinetix_imaging(t, close_all_shutters=True)
+        return t
 
-    def tweezer_modulation(self, t, label='sine'):
-        pass
+    def do_kinetix_imaging(self, t, close_all_shutters=False):
```

Comment:
Is there a better name for this function? The name do_kinetix_imaging sounds like it should belong in the Camera hardware class.

Reply:
replace with do_tweezer_imaging